### ISSUE TITLE
Run Shortcodes in Descriptions options don't work correctly

### ISSUE THREAD

**Issue by wpsmort** [semperfiwebdesign/all-in-one-seo-pack#1041]
This was originally reported under #943.
Related to #1035 and #787.

In #943, Angus reported that text wrapped in brackets is not output in og:description tags.  I tested this and confirmed that it affects meta descriptions and og:descriptions.  We attempted a fix for this in version 2.3.15 but had to roll it back due to numerous bug reports from users.

The two Run Shortcodes In Autogenerated Descriptions options need to be fixed so that these work and then users can choose whether to run shortcodes or not.

**Comment by wpsmort** [comment]
To reproduce this, create a post and enter the following as the Title, Content and Excerpt:

Post with [brackets] in it

Check the source code and you'll see that the title tag and og:title is correct but the meta description and og:description looks like this:

"Post with in it"

Testing this with the Run Shortcodes options disabled or enabled makes no difference.  

We need to account for sites where users have posts/pages with shortcodes in as well as posts/pages where they have text that contains brackets.

**Comment by amostajo** [comment]
@wpsmort So are we going to rollback to the functionality found prior 2.3.15?

**Comment by wpsmort** [comment]
We should not change behavior for auto generated descriptions, instead we need to verify that 1). the behavior in the current release is the same as in releases prior to v2.3.15 and 2). that nothing is broken with auto generated descriptions. 

We do need to make sure that if a user enters text with brackets in the SEO Description field or OG Description field in the AIOSEOP meta box on a post then we output that as text. There's no reason why someone would enter a shortcode in our Description fields. 

The goal here is to fix the original issue #943 without changing behavior for users who have Autogenerate Descriptions enabled.

**Comment by michaeltorbert** [comment]
Since we're back to pre-2.3.15 functionality with this (verify this @wpsmort ), I'm moving this to 2.3.17.

**Comment by wpsmort** [comment]
@michaeltorbert I have tested and can confirm that the current live release v2.3.15.3 is back to pre-2.3.15 functionality.  

I recommend that we close this issue as the problem is not with the Run Shortcodes options, those options work fine and we've confirmed this by reviewing the code.

We should leave issue #1035 open and assign it to 2.3.17 and rename it Improve handling of shortcodes in manually entered descriptions.  This is because we don't intend to change anything with autogenerated descriptions, we just want to fix a bug with outputting shortcodes manually entered in the SEO and OG description fields.
